x-pack/test/security_solution_cypress/cypress/support/es_client.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { systemIndicesSuperuser } from '@kbn/test';
+import { createEsClient } from '@kbn/security-solution-plugin/scripts/endpoint/common/stack_services';
+
+export const esClient = (
+  on: Cypress.PluginEvents,
+  config: Cypress.PluginConfigOptions
+): Promise<void> => {
+  const isServerless = config.env.IS_SERVERLESS;
+  const isCloudServerless = config.env.CLOUD_SERVERLESS;
+
+  const user = {
+    username: config.env.ELASTICSEARCH_USERNAME,
+    password: config.env.ELASTICSEARCH_PASSWORD,
+  };
+
+  const authOverride = isServerless ? (isCloudServerless ? user : systemIndicesSuperuser) : user;
```

Comment:
Would like to know what's the difference between user and systemIndicesSuperuser?

Reply:
`system_indices_superuser` is a user created for testing purposes (an operator one) that does not have restrictions, that user is the one used on ESS and stateless environments to access internal indexes directly and does not exist on MKI environments.

Reply:
maybe this explanation can be added as comment here?